base/compiler/ssair/inlining.jl

```diff
@@ -399,7 +411,8 @@ function ir_inline_item!(compact::IncrementalCompact, idx::Int, argexprs::Vector
             # face of rename_arguments! mutating in place - should figure out
             # something better eventually.
             inline_compact[idx′] = nothing
-            stmt′ = ssa_substitute!(idx′, stmt′, argexprs, sig, sparam_vals, sp_ssa, linetable_offset, boundscheck, inline_compact)
+            insert_node_before! = new_inst->insert_node!(inline_compact, SSAValue(idx′), new_inst)
```

Comment:
Can we handle closure capture of `inline_compact` here? I thought that because semantically it has 2 assignments we would conservatively assume it may get reassigned later

Reply:
I'll check.